### PR TITLE
Wraps `lowertext()` to ensure proper stringification.

### LINKED ISSUE #82440: OpenDream Cleanup: Several bad `lowertext()` calls
## Issue Summary

BYOND's string procs do not stringify `src`.
image

Just do a global search for `lowertext(src)`, but here's the ones OD found:
```
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:487:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:493:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:501:30: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:652:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:657:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:663:30: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:758:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:764:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/food_reagents.dm:769:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/other_reagents.dm:328:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/other_reagents.dm:332:31: lowertext(...) argument "T": Invalid var value type "null", expected "text"
Warning OD2702 at code/modules/reagents/chemistry/reagents/other_reagents.dm:339:30: lowertext(...) argument "T": Invalid var value type "null", expected "text"
```

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Fixes #82440

This PR just creates a new macro, `LOWER_TEXT()` (yes the irony is not lost on me) to wrap around all calls of `lowertext()` and ensure that whatever we input into that proc will be stringified using the `"[]"` (or `tostring()` for the nerds) operator. very simple.

I also added a linter to enforce this (and prevent all forms of regression) because I think that machines should do the menial work and we shouldn't expect maintainers to remember this, let me know if you disagree. if there is a time when it should be opted out for some reason, the linter does respect it if you wrap your input with the `UNLINT()` function.

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
~~could just be lower_text()~~
❤️

**Comment by san7890** [comment]
> ~could just be lower_text()~ ❤️

it's not uncommon for text-macros to be lower-case (precedent being the `span_x()` stuff) so fuck it we ball

**Comment by san7890** [comment]
image

what the spruce? i'm not particularly fond of having to rename var names if we operate in this system but lmk what you think

**Comment by LemonInTheDark** [comment]
bro wut

**Comment by ZephyrTFA** [comment]
ah Byond, where a macro is global even without parenthesis

File: code/__DEFINES/text.dm [hunk 1]
```diff
@@ -64,6 +64,10 @@
  */
 #define PREVENT_CHARACTER_TRIM_LOSS(integer) (integer + 1)
 
+/// BYOND's string procs, `lowertext()` here, don't support being used on datum references (as in it doesn't look for a name for stringification)
+/// We just use this macro to ensure that we will only pass strings to this BYOND-level function without developers needing to really worry about it.
+#define lower_text(thing) lowertext(UNLINT("[thing]"))
```
[the remark this answers is not in the document]
Comment (san7890) [reply code/__DEFINES/text.dm:69]: i concur on retrospect

**Comment by san7890** [comment]
the grep is fugly (grepping the grep results) but i'm ultimately okay with it

**Pull request merged**